policies/branch-policy.md

```diff
@@ -0,0 +1,88 @@
+Branch Policy
+=============
+
+The openssl repository contains the following maintained branches:
+
+### The master branch
+
+- Any type (bug fix, feature, refactoring, ...) of pull requests is allowed.
+- The development of the next minor or major release happens there.
+  Which kind of release it is is defined by OMC at some point after the
+  master branch is opened after the previous release was branched.
+  Implicitly it means that any API/ABI breaking changes are not allowed
+  on the master branch unless OMC defines the next release will be a major
+  release.
+- Any changes merged to the master branch must be ported to the next major
+  and future minor branches if they are applicable. This can be done by
+  directly cherry-picking the changes when merging if there are no conflicts.
+  By this we must ensure that no features or bug fixes are unintentionally
+  lost in future major or minor releases.
+
+### The supported release branches
+
+- Only bug fixes and documentation changes are allowed.
+- The development of the next patch releases of supported minor releases
+  happens there.
+- By exception given by OMC also other types of pull requests can be merged.
+- During the final year of support, we do not commit to anything other than
+  security fixes.
+- Bug fix and documentation pull requests must be always merged to the
+  latest branch (including the master branch) where the bug or documentation
+  deficiency is present. It can be then merged (backported or directly
+  cherry-picked) to all older branches where the deficiency is present.
+
+### A next major branch
+
+- The development of the next major release happens there. Implicitly it
+  means that any API/ABI breaking changes are allowed but OMC can (and
+  usually will) limit the extent of the breakage allowed.
+- Major features are allowed. Examples of a major feature: A completely
+  new implementation of a protocol. New API for pluggable crypto modules.
+- Major refactoring is allowed. Examples of major refactoring: Splitting
+  libcrypto into multiple hierarchically dependent libraries.
+- The branch is identical to the master branch in case the next release
+  in development is a major release.
```

Comment:
In this case do we really need this branch? It seems redundant in this scenario

Reply:
It's true that we would have to somehow keep it in sync with the master branch. However it makes the source git more predictable to outside contributors I think.

Reply:
Hmm, depends...  as long as we do like now, with master being the development of the next release, major or minor alike, then I don't see the point.  What this suggests is that we would have multiple development branches...  and I can't recall how well we've discussed that.

Reply:
> What this suggests is that we would have multiple development branches... and I can't recall how well we've discussed that.

We discussed it that we want it the last time we discussed the policy proposal on OTC meeting. So we can merge stuff to the future-major or future-minor branches and not keep open PRs for indefinite amount of time. But yeah, further discussion is needed for sure.

Reply:
IIRC the need for a future-major branch when the next release is going to be a major release is supposed to exist for those PRs that OMC decides are out of scope for the next release and are deferred to the following major release.

In this case this item needs to be rewritten, as this branch and master might be out of sync in such a scenario.